### ISSUE TITLE
Attachments' names are "tcmime.xxxx.xxxx.xxxx.bin"

### ISSUE THREAD

**Issue by guow93** [jstedfast/MailKit#277]
If the attachment of a email has a  file name which contains Chinese or non-ASCII characters and have a long length, no matter what  the QQ mail is, sender or receiver,  the mail's attachment in QQ mail's sent or in box  shows as "tcmime.xxxx.xxxx.xxxx.bin", but other mail servers perform normal. You can see something about this in https://support.microsoft.com/en-us/kb/2402064

**Comment by guow93** [comment]
I used "using BodyBuild" and "cast a MailMessage to a MimeMessage"

**Comment by jstedfast** [comment]
The mail client you are using to read the message does not support rfc2231 encoding. Most likely it expects the filename parameter to be encoded using the rfc2047 encoding (which as never legal to use for encoding parameter values... but *sigh*, some mail software just sucks).

There are now 2 ways of controlling the encoding method used for parameter values.

The first way is to set the encoding method on each individual Parameter:

```csharp
param.EncodingMethod = ParameterEncodingMethod.Rfc2047;
```

The second way is to set the default parameter encoding method on the FormatOptions used for writing out the message and/or MIME part(s):

```csharp
var options = FormatOptions.Default.Clone ();
options.ParameterEncodingMethod = ParameterEncodingMethod.Rfc2047;

message.WriteTo (options, stream);
```

**Comment by guow93** [comment]
I have seen the similar question--"Why do attachments with unicode filenames appear as "ATT0####.dat" in Outlook?" in http://www.mimekit.net/docs/html/FrequentlyAskedQuestions.htm#UntitledAttachments. I just don't know how to use the first way when I use BodyBuilder to add attachments to form a MineMessage, but now, I have tried the first way and get it. 
for(int i = 0; i < builder.Attachments.Count<MimeEntity>(); i++)
{
     Parameter param;
     if (builder.Attachments.ElementAt<MimeEntity(i).ContentDisposition.Parameters.TryGetValue("filename", out param))
   param.EncodingMethod = ParameterEncodingMethod.Rfc2047;
}
I succeed through these sentences, is it right?
